src/utils.rs

```diff
@@ -22,16 +22,18 @@ pub(crate) async fn check_cookie<'a, F, Fut, R>(
 	callback: F,
 ) -> Result<R, Status>
 where
-	F: FnOnce(&'a str, &'a str) -> Fut,
+	F: FnOnce(&'a str, &'a str, &'a str) -> Fut,
 	Fut: std::future::Future<Output = Result<R, Status>>,
 {
 	let username = cookies.get("username");
 	let seed = cookies.get("seed");
-	match (username, seed) {
-		(Some(username_cookie), Some(seed_cookie)) => {
+	let object_id = cookies.get("object_id");
+	match (username, seed, object_id) {
```

Comment:
I haven't run this code, but I think something like this could work here:
``` rust
let username = cookies.get("username").ok_or(Status::Forbidden)?.value();
let seed =  cookies.get("seed").ok_or(Status::Forbidden)?.value();
let object_id = cookies.get("object_id").ok_or(Status::Forbidden)?.value();
callback(username, seed, object_id).await
```